cyber/record/file/record_file_writer.h

```diff
@@ -123,7 +123,10 @@ bool RecordFileWriter::WriteSection(const T& message) {
     AERROR << "Do not support this template typename.";
     return false;
   }
-  Section section = {type, message.ByteSize()};
+  Section section;
+  /// zero out whole struct even if padded
+  memset(&section, 0, sizeof(section));
+  section = {type, static_cast<int64_t>(message.ByteSize())};
```

Comment:
Not need static cast to int64_t here.

Reply:
According to https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message_lite#MessageLite.ByteSize, `ByteSize()` returns `int`. `int` doesn't specify a size. On my system `int` is 4 bytes. I think we still want the cast.

Reply:
Converting low precision int to high precision int do not need explicit type casting. 
I think it's better to use explicit casting where it is necessary. 
The compiler will issue an error message unless an explicit cast is made.

Reply:
That works for me. I made the change.